File: frontend/src/types/matter.ts
export type MatterStatus = "active" | "pending" | "closed" | "archived";

export type PractitionerRole = "admin" | "lawyer" | "staff";

export type MatterListFilter = "all" | "active" | "pending" | "closed" | "archived";

export type MatterTabKey =
  | "overview"
  | "tasks"
  | "calendar"
  | "documents"
  | "billing"
  | "notes"
  | "audit";

export interface MatterKpi {
  label: string;
  value: string;
  subtext: string;
  tone: "default" | "warning" | "success" | "danger";
}

export interface MatterTimelineItem {
  id: string;
  title: string;
  detail: string;
  at: string;
}

export interface MatterWorkItem {
  id: string;
  title: string;
  owner: string;
  dueLabel: string;
}

export interface MatterListItem {
  id: string;
  reference: string;
  title: string;
  clientName: string;
  matterType: string;
  practiceArea: string;
  status: MatterStatus;
  leadLawyerName: string;
  nextDeadlineAt: string | null;
  unpaidBalanceGhs: number;
  assignedToMe: boolean;
}

export interface MatterWorkspaceStats {
  activeCount: number;
  pendingCount: number;
  dueThisWeekCount: number;
  outstandingBalanceGhs: number;
}

export interface UseMattersResult {
  matters: MatterListItem[];
  stats: MatterWorkspaceStats;
  isLoading: boolean;
  isError: boolean;
}

export interface MatterDetail {
  id: string;
  reference: string;
  title: string;
  clientName: string;
  matterType: string;
  practiceArea: string;
  status: MatterStatus;
  court: string | null;
  suitNumber: string | null;
  opposingParty: string | null;
  opposingCounsel: string | null;
  leadLawyerName: string | null;
  nextDeadlineAt: string | null;
  openedAt: string;
  unpaidBalanceGhs: number;
  recentActivitySummary: string;
  recentActivityCount: number;
  summary: string;
  focusNote: string;
  recentTimeline: MatterTimelineItem[];
  outstandingWork: MatterWorkItem[];
}

File: backend/app/schemas/matter.py
"""Pydantic schemas for the starter read-only matter endpoints."""

from pydantic import BaseModel


class MatterTimelineItem(BaseModel):
    id: str
    title: str
    detail: str
    at: str


class MatterWorkItem(BaseModel):
    id: str
    title: str
    owner: str
    dueLabel: str


class MatterListItem(BaseModel):
    id: str
    reference: str
    title: str
    clientName: str
    matterType: str
    practiceArea: str
    status: str
    leadLawyerName: str
    nextDeadlineAt: str | None
    unpaidBalanceGhs: float
    assignedToMe: bool


class MatterDetail(MatterListItem):
    court: str | None
    suitNumber: str | None
    opposingParty: str | None
    opposingCounsel: str | None
    leadLawyerName: str | None
    openedAt: str
    recentActivitySummary: str
    recentActivityCount: int
    summary: str
    focusNote: str
    recentTimeline: list[MatterTimelineItem]
    outstandingWork: list[MatterWorkItem]
